xarray/core/datatree.py

```diff
@@ -158,6 +158,19 @@ def _check_alignment(
             _check_alignment(child_path, child_ds, base_ds, child.children)
 
 
+def _check_for_slashes_in_names(variables: Iterable[Hashable]) -> None:
+    offending_variable_names = [
+        name for name in variables if isinstance(name, str) and "/" in name
+    ]
+    if len(offending_variable_names) > 0:
+        raise KeyError(
```

Comment:
This should be a `ValueError` not a `KeyError`

```suggestion
        raise ValueError(
```